### DISCUSSION TITLE
Changing SSL Ciphers? (CHACHA20)

### DISCUSSION THREAD

**Discussion by idle-user** [NginxProxyManager/nginx-proxy-manager#1842] [Q&A]
Maybe it's not the job of this proxy manager, but is there a way to change the SSL Cipher to CHACHA20? 

I've tried mounting `/etc/nginx/conf.d/` and updating `includes/ssl-ciphers.conf` to
```
ssl_protocols               TLSv1.2 TLSv1.3;
ssl_ciphers                 ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-CHACHA20-POLY1305;
ssl_prefer_server_ciphers   on;
```

But the security page via browser still shows AES even after restarts. Maybe there's something I'm missing.

Any advice?
Thank you,

**Comment by mochman** [comment]
I'm not sure what's going on with your setup, but I've dome something similar and the ciphers have changed for me.
Instead of mounting the `/etc/nginx/conf.d/ ` folder, I mounted the file.
```
volumes:
  - ./sslciphers.conf:/etc/nginx/conf.d/include/ssl-ciphers.conf
 ```

**Reply by idle-user** [reply]
I'll give this a shot when I get a chance and will provide a follow up. 

May I see a copy of your .conf just in case?
Thank you

**Reply by mochman** [reply]
```
ssl_session_timeout 5m;
ssl_session_cache shared:SSL:50m;

# intermediate configuration. tweak to your needs.
ssl_protocols TLSv1.2 TLSv1.3;
ssl_ciphers 'ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256';
ssl_prefer_server_ciphers on;

add_header X-Robots-Tag "noindex, nofollow, nosnippet, noarchive";
```
